src/main/java/co/cask/hydrator/plugin/source/KafkaConfig.java

```diff
@@ -237,21 +237,21 @@ public Schema getMessageSchema() {
         "Schema must contain at least one other field besides the time and key fields.");
     }
 
-    if (getTimeField() != null && !timeFieldExists) {
+    if (!getTimeField().equals("timestamp") && !timeFieldExists) {
```

Comment:
Should the other fields like offset and key also have that part of the condition statement removed?